api/spec/controllers/spree/api/v1/taxons_controller_spec.rb

```diff
@@ -31,18 +32,31 @@ module Spree
       end
 
       it 'paginates through taxons' do
-        new_taxon = create(:taxon, name: 'Go', taxonomy: taxonomy, parent_id: taxonomy.taxons.first.id)
+        new_taxon = create(:taxon, name: 'Go', taxonomy: taxonomy, parent_id: taxonomy.root.id)
         taxonomy.root.children << new_taxon
-        expect(taxonomy.root.children.count).to eql(2)
+        expect(taxonomy.root.children.count).to eql(3)
         api_get :index, taxonomy_id: taxonomy.id, page: 1, per_page: 1
         expect(json_response['count']).to eql(1)
-        expect(json_response['total_count']).to eql(2)
+        expect(json_response['total_count']).to eql(3)
         expect(json_response['current_page']).to eql(1)
         expect(json_response['per_page']).to eql(1)
-        expect(json_response['pages']).to eql(2)
+        expect(json_response['pages']).to eql(3)
       end
 
       describe 'searching' do
+        context 'within a taxonomy' do
+          before do
+            api_get :index, taxonomy_id: taxonomy.id, q: { name_cont: name }
+          end
+
+          context 'searching for top level taxon' do
```

Comment:
You don't test if it searched correctly. This test would pass even if api returned `rust_taxon` as second element in array.